admin/authentication/nca-tools-authentication-strategy.js

```diff
@@ -29,36 +31,51 @@ module.exports = async function (req, done) {
   /**
    * Store the logon attempt
    */
-  const logonEvent = new AdminLogonEvent({
+  const logonEvent = {
     sessionId: req.session.id,
     body: JSON.stringify(req.body),
     remoteIp: (req.headers['x-forwarded-for'] || req.connection.remoteAddress),
     userAgent: req.headers['user-agent'],
     loginMethod: 'nca-tools'
-  })
+  }
 
   try {
     const senderPublicKey = config.TSO_AUTH_PUBLIC_KEY
     const recipientPrivateKey = config.MTC_AUTH_PRIVATE_KEY
     const userData = await ncaToolsAuthService(encKey, encIv, encData, encSignature, senderPublicKey, recipientPrivateKey)
 
+    let userRecord = await userDataService.sqlFindOneByIdentifier(userData.UserName)
+    if (!userRecord) {
+      const mtcRoleName = roleService.mapNcaRoleToMtcRole(userData.UserType)
```

Comment:
Should this code be in the user auth strategy or a service?

Reply:
Yes.  I've been looking through all the associated code and am considering structural changes once I have the POC working.